### PR TITLE
data_collector_update_for_web_interface_call 

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/OctoBot-Trading/pull/69
requires https://github.com/Drakkar-Software/OctoBot-Trading/pull/71

### CODE REVIEW

File: octobot_backtesting/api/exchange_data_collector.py [hunk 1]
```diff
@@ -13,15 +13,15 @@
...
 
-async def collect_exchange_historical_data(config, exchange_name, symbols, time_frames=None):
-    collector_class = get_single_deepest_child_class(ExchangeDataCollector)
-    collector_instance = collector_class(config, exchange_name, symbols, time_frames)
-    if time_frames is None:
-        collector_instance.use_all_available_timeframes()
-    collector_instance.initialize()
+async def collect_exchange_historical_data(exchange_name, symbols, time_frames=None):
+    collector_class = get_single_deepest_child_class(AbstractExchangeHistoryCollector)
+    collector_instance = collector_class({}, exchange_name, symbols, time_frames,
```
Comment (Herklos) [inline octobot_backtesting/api/exchange_data_collector.py]: :+1:

File: octobot_backtesting/collectors/data_collector.py [hunk 2]
```diff
@@ -16,33 +16,36 @@
...
 from octobot_backtesting.data.database import DataBase
+from octobot_backtesting.data.data_file_manager import get_file_ending
 from octobot_backtesting.importers.data_importer import DataImporter
 
 
 class DataCollector:
     IMPORTER = DataImporter
 
-    def __init__(self, config, path=BACKTESTING_FILE_PATH):
+    def __init__(self, config, path=BACKTESTING_FILE_PATH, data_format=DataFormats.REGULAR_COLLECTOR_DATA):
```
Comment (Herklos) [inline octobot_backtesting/collectors/data_collector.py]: :+1:

File: octobot_backtesting/collectors/exchanges/exchange_collector.pxd [hunk 3]
```diff
@@ -23,3 +23,7 @@ cdef class ExchangeDataCollector(DataCollector):
 
     cdef public list symbols
     cdef public list time_frames
+    cdef public bint use_all_available_timeframes
```
Comment (Herklos) [inline octobot_backtesting/collectors/exchanges/exchange_collector.pxd]: I don't think this have to be public.

File: octobot_backtesting/data/data_file_manager.py [hunk 4]
```diff
@@ -14,58 +14,15 @@
...
-from octobot_commons.symbol_util import merge_currencies
-from octobot_commons.time_frame_manager import find_min_time_frame
+from octobot_commons.enums import TimeFrames
 
-from octobot_backtesting.constants import BACKTESTING_TIME_FRAMES_TO_DISPLAY, BACKTESTING_DATA_FILE_TIME_DISPLAY_FORMAT, \
-    BACKTESTING_DATA_FILE_TIME_READ_FORMAT, BACKTESTING_DATA_FILE_EXT, BACKTESTING_DATA_FILE_TIME_WRITE_FORMAT
-from octobot_backtesting.enums import DataFormatKeys, DataFormats
-
-
-def interpret_file_name(file_name):
```
Comment (Herklos) [inline octobot_backtesting/data/data_file_manager.py]: These methods are not used anymore right ?

File: octobot_backtesting/data/data_file_manager.py [hunk 5]
```diff
@@ -78,62 +35,33 @@ def get_file_ending(data_type):
...
-    time_object = time.strptime(time_info, BACKTESTING_DATA_FILE_TIME_READ_FORMAT)
-    return time.strftime(BACKTESTING_DATA_FILE_TIME_DISPLAY_FORMAT, time_object)
-
-
-def get_file_description(data_collector_path, file_name):
-    exchange_name, symbol, time_info, data_type = interpret_file_name(file_name)
+    return datetime.fromtimestamp(time_info).strftime(BACKTESTING_DATA_FILE_TIME_DISPLAY_FORMAT)
+
+
+async def get_database_description(database):
```
Comment (Herklos) [inline octobot_backtesting/data/data_file_manager.py]: :+1:

File: octobot_backtesting/data/data_file_manager.py [hunk 6]
```diff
@@ -78,62 +35,33 @@ def get_file_ending(data_type):
...
+
+async def get_file_description(database):
+    description = await get_database_description(database)
     return {
-        DataFormatKeys.SYMBOL.value: symbol,
-        DataFormatKeys.EXCHANGE.value: exchange_name,
-        DataFormatKeys.DATE.value: get_date(time_info),
-        DataFormatKeys.CANDLES.value: get_number_of_candles(join(data_collector_path, file_name)),
-        DataFormatKeys.TYPE.value: data_type.name
+        DataFormatKeys.SYMBOLS.value: description[DataFormatKeys.SYMBOLS.value],
```
Comment (Herklos) [inline octobot_backtesting/data/data_file_manager.py]: :100:

File: octobot_backtesting/importers/exchanges/exchange_importer.py [hunk 7]
```diff
@@ -37,13 +38,10 @@ def __init__(self, config, file_path):
         await self.database.initialize()
 
         # load description
-        description = (await self.database.select(DataTables.DESCRIPTION, size=1))[0]
-
-        version = description[1]
-        if version == "1.0":
```
Comment (Herklos) [inline octobot_backtesting/importers/exchanges/exchange_importer.py]: Why removing description versionning support ?

File: octobot_backtesting/data/data_file_manager.py (hunk 4 above)
Comment (GuillaumeDSM) [reply octobot_backtesting/data/data_file_manager.py]: Not for now, we might want a similar code when coding converters from v3 backtesting data but I don't think it should be in this module anyway.

File: octobot_backtesting/importers/exchanges/exchange_importer.py (hunk 7 above)
Comment (GuillaumeDSM) [reply octobot_backtesting/importers/exchanges/exchange_importer.py]: It's now in get_database_description:
image

**Pull request merged**